pkg/apply/prune/prune_test.go

```diff
@@ -784,88 +776,12 @@ func TestPruneWithErrors(t *testing.T) {
-			prevInventory: []*unstructured.Unstructured{},
-			expectedObjs:  []*unstructured.Unstructured{},
-		},
-		"no local objects, with inventory equals all prune objs": {
-			localObjs:     []*unstructured.Unstructured{},
-			prevInventory: []*unstructured.Unstructured{pod, pdb, namespace},
-			expectedObjs:  []*unstructured.Unstructured{pod, pdb, namespace},
-		},
-		"set difference equals one prune object": {
-			localObjs:     []*unstructured.Unstructured{pod, pdb},
-			prevInventory: []*unstructured.Unstructured{pdb, namespace},
-			expectedObjs:  []*unstructured.Unstructured{namespace},
-		},
-		"local and inventory the same equals no prune objects": {
-			localObjs:     []*unstructured.Unstructured{pod, pdb},
-			prevInventory: []*unstructured.Unstructured{pod, pdb},
-			expectedObjs:  []*unstructured.Unstructured{},
-		},
-		"two prune objects": {
-			localObjs:     []*unstructured.Unstructured{pdb},
-			prevInventory: []*unstructured.Unstructured{pod, pdb, namespace},
-			expectedObjs:  []*unstructured.Unstructured{pod, namespace},
-		},
-		"skip pruning objects whose resource types are unrecognized by the cluster": {
-			localObjs:     []*unstructured.Unstructured{pdb},
-			prevInventory: []*unstructured.Unstructured{testutil.Unstructured(t, crontabCRManifest), pdb, namespace},
-			expectedObjs:  []*unstructured.Unstructured{namespace},
-		},
-		"local objs, inventory disjoint means inventory is pruned": {
-			localObjs:     []*unstructured.Unstructured{pdb},
-			prevInventory: []*unstructured.Unstructured{pod, namespace},
-			expectedObjs:  []*unstructured.Unstructured{pod, namespace},
-		},
-	}
-	for name, tc := range tests {
-		t.Run(name, func(t *testing.T) {
-			objs := make([]runtime.Object, 0, len(tc.prevInventory))
-			for _, obj := range tc.prevInventory {
-				objs = append(objs, obj)
-			}
-			po := Pruner{
-				InvClient: inventory.NewFakeClient(object.UnstructuredSetToObjMetadataSet(tc.prevInventory)),
-				Client:    fake.NewSimpleDynamicClient(scheme.Scheme, objs...),
-				Mapper: testrestmapper.TestOnlyStaticRESTMapper(scheme.Scheme,
-					scheme.Scheme.PrioritizedVersionsAllGroups()...),
-			}
-			currentInventory := createInventoryInfo(tc.prevInventory...)
-			actualObjs, err := po.GetPruneObjs(currentInventory, tc.localObjs, Options{})
-			if err != nil {
-				t.Fatalf("unexpected error %s returned", err)
-			}
-			if len(tc.expectedObjs) != len(actualObjs) {
-				t.Fatalf("expected %d prune objs, got %d", len(tc.expectedObjs), len(actualObjs))
-			}
-			actualIds := object.UnstructuredSetToObjMetadataSet(actualObjs)
-			expectedIds := object.UnstructuredSetToObjMetadataSet(tc.expectedObjs)
-			if !object.ObjMetadataSetEquals(expectedIds, actualIds) {
-				t.Errorf("expected prune objects (%v), got (%v)", expectedIds, actualIds)
+				events = append(events, e)
 			}
+			receivedEvents := testutil.EventsToExpEvents(events)
+			testutil.AssertEqual(t, tc.expectedEvents, receivedEvents)
```

Comment:
This test cleanup to use testutil.AssertEqual and testutil.EqualErrorString could be pulled out into another PR.